sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -186,6 +187,14 @@ case class DataSource(
         val path = caseInsensitiveOptions.getOrElse("path", {
           throw new IllegalArgumentException("'path' is not specified")
         })
+        if (!sparkSession.conf.get(SQLConf.STREAMING_SCHEMA_INFERENCE) &&
```

Comment:
This and the other 2 lines: could you please define a local `val` and name it in a way that tells what the `if` is supposed to guard?